tests/zquantum/qaoa/recursive_qaoa_test.py

```diff
@@ -360,14 +430,13 @@ def test_compatability_with_x_ansatz(
 
         recursive_qaoa = RecursiveQAOA(
             1,
+            hamiltonian,
             x_ansatz,
-            initial_params,
-            optimizer,
-            estimation_tasks_factory_generator,
-            cost_function_factory,
+            inner_optimizer,
         )
 
-        solutions = recursive_qaoa(hamiltonian)
+        opt_result = recursive_qaoa.minimize(cost_function_factory, initial_params)
+        solutions: List[Tuple[int]] = opt_result.opt_solutions
```

Comment:
I think you're missing contracts from NestedOptimizer?

Reply:
I think the contracts are [here](https://github.com/zapatacomputing/z-quantum-qaoa/pull/78/files#diff-9ba2a7a0fdd953c36cb5570103b0336eaf6e7ea08bc9275bc5202c50fcad7d88R95)

Reply:
Thanks I missed that!